client/sites/plan/components/current-plan-panel/index.tsx

```diff
@@ -120,7 +104,7 @@ export default function CurrentPlanPanel() {
 				</div>
 			</div>
 
-			<PlanStats />
+			<PlanStats needMoreStorageTracksEventName="calypso_plans_need_more_storage_click" />
```

Comment:
This technically leaks implementation details to the parent. 

Does it make more sense to pass the context down to the children instead?

`<PlanStats context="hosting_overview" />`

...
```
<NeedMoreStorage
   noLink={ footerWrapperIsLink }
   tracksEventName={`calypso_hosting_${context}_need_more_storage_click` }
   onClick={ () => setIsStorageAddOnsModalOpen( true ) }
/>
```

Reply:
That's ideal, but I remember somewhere that we need to type the Tracks event strings verbatim in the codebase. This is so that we can grep Tracks events easily in the codebase. Let me search for docs...

Reply:
> Don’t programmatically create event names. They should be hardcoded so they are easily searchable in the code base.

PCYsg-4S2-p2

Reply:
Thanks for the learn-up. That makes sense.

For this case, we'd probably need to over-engineer something to stay SOLID. 🤔

Reply:
I guess you could use an object for the lookup to preserve the string.

Not a strong opinion.

Reply:
Yeah, the thing with Tracks events is... there's this tradeoff that parent needs to know the individual "subobjects" that we want to track downstreams. A similar pattern here: https://github.com/Automattic/wp-calypso/blob/3d56a33343aed8fdc91bf7b9c765eb860bcb61a0/client/a8c-for-agencies/sections/overview/sidebar/quick-links/index.tsx. In any case, whether it's object/array/simple props, the parent need to know somehow, so I opted for the simplest way 👍 

Thanks for the review!